client/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+pub mod util;
+
+use crate::util::*;
+use ed25519_dalek::{Keypair, PublicKey};
+use hpos_config_core::admin_keypair_from;
+use serde::*;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct HpAdminKeypair(Keypair);
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Payload {
+    method: String,
+    request: String,
+    body: String,
+}
+
+#[wasm_bindgen]
+impl HpAdminKeypair {
+    /// @description Create an Ed25519 keypair for API signing calls to HPOS
+    /// @example
+    /// myKeys = new HpAdminKeypair( hc_public_key_string, email, password );
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        hc_public_key_string: String,
+        email: String,
+        password: String,
+    ) -> Fallible<HpAdminKeypair> {
+        console_error_panic_hook::set_once();
+        let keypair = new_inner(hc_public_key_string, email, password)?;
+        Ok(Self(keypair))
+    }
+
+    /// @description Sign payload and return base64 encoded signature.
+    /// Requires properly formatted payload:
+    /// const payload = {
+    ///     method: String,
+    ///     request: String,
+    ///     body: String
+    /// }
+    /// @example
+    /// myKeys = new HpAdminKeypair( hc_public_key_string, email, password );
+    /// const payload = {
+    ///     method: "get",
+    ///     request: "/someuri",
+    ///     body: ""
+    /// }
+    /// myKeys.sign( payload );
+    #[wasm_bindgen]
+    pub fn sign(&self, payload: &JsValue) -> Fallible<String> {
+        // return meaningful error message via JsValue
+        let payload_vec = parse_payload(payload)?;
+        let signature = self.0.sign(&payload_vec);
+        Ok(base64::encode_config(
+            &signature.to_bytes()[..],
+            base64::STANDARD_NO_PAD,
+        ))
+    }
+}
+
+fn new_inner(hc_public_key_string: String, email: String, password: String) -> Fallible<Keypair> {
```

Comment:
`new_inner` still wasn't collapsed into `new`, which should've been addressed as part of #6.